modules/custom/d_commerce_products_list/src/Plugin/Block/CommerceMobileFiltersSubmit.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Drupal\d_commerce_products_list\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Provides a 'CommerceMobileFiltersSubmit' Block.
+ *
+ * @Block(
+ *   id = "commerce_mobile_filters_submit",
+ *   admin_label = @Translation("Commerce Mobile filters submit"),
+ *   category = @Translation("Facets"),
+ * )
+ */
+class CommerceMobileFiltersSubmit extends BlockBase {
```

Comment:
Ten form jest bardzo zbliżony do CommerceMobileFilters, warto by pomyśleć nad takim refactoringiem, aby wydzielić czesci wspólne i wynieść do do nadrzędnej klasy.
Różnice można zdefiniować we właściowościach, lub zdefiniowanych metodach, ktore wykonują niezbędne dodatkowe operacje/modyfikacje np. na render arrayu.